mantidproject/mantid
# Change column format printed in header depending on modulation vector…

Bug fixed - the column formats in the header did not take into account the presence of extra columns if the peak workspace contained satellite peaks created with a modulation vector (with PredictFractionalPeaks or PredictSatellitePeaks).

**To test:**
1) Run script that loads in some parent peaks (.nxs contained in  [TOPAZ_peaks.zip](https://github.com/mantidproject/mantid/files/4286434/TOPAZ_peaks.zip)) and then predicts fractional with 1 and 2 modulation vectors and exports all to jana format.
 

    LoadNexus(Filename='TOPAZ_3132_peaks.nxs',  OutputWorkspace='peaks')
    SaveReflections('peaks',Filename='peaks_jana.hkl',Format='Jana',SplitFiles=True)
    PredictFractionalPeaks(Peaks='peaks', RequirePeaksOnDetector=False, ModVector1='-0.5,0,0.5', 
        MaxOrder=1, FracPeaks='frac')
    SaveReflections('frac',Filename='frac_1q_jana.hkl',Format='Jana',SplitFiles=True)
    PredictFractionalPeaks(Peaks='peaks', RequirePeaksOnDetector=False, ModVector1='-0.5,0,0.5', 
        ModVector2='0,0.2,0',MaxOrder=1, FracPeaks='frac_both')
    SaveReflections('frac_both',Filename='frac_2q_split_jana.hkl',Format='Jana',SplitFiles=True)
    SaveReflections('frac_both',Filename='frac_2q_together_jana.hkl',Format='Jana',SplitFiles=False)

2) Check the header line contains the right column formatting (see the original issue #28183)

Fixes #28183

*This does not require release notes* because the bug was introduced in this release


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
